x-pack/legacy/plugins/ml/public/application/explorer/actions/job_selection.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { from } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { mlFieldFormatService } from '../../services/field_format_service';
+import { mlJobService } from '../../services/job_service';
+
+import { ExplorerAppState } from '../reducers';
+import { createJobs, restoreAppState } from '../explorer_utils';
+
+export function jobSelectionActionCreator(
```

Comment:
I don't get why do you need to return an Observable here? later you pass it to `explorerAction$` which received both Actions and ActionPayloads as Observable 🤔

Reply:
It's inspired by how side-effects are implemented in redux: If it's a plain action object it gets passed on to the reducer, if it's a function (e.g. `redux-thunk`) or an observable (`redux-observable`) it gets treated like a side-effect (the side-effect will then later on return a plain action with new data to trigger the reducer). In some earlier version of the code I triggered actions that had observables as the payload. That's no longer used so I cleaned up that bit here: 60a4670.